### PR TITLE
feat: add rpc traceBlock

### PR DESCRIPTION
**Pull request by tdelabro**
Add `traceBlockTransactions` rpc 

supersed #1411

### CODE REVIEW

File: crates/client/rpc/src/trace_api.rs [hunk 1]
```diff
@@ -106,6 +170,17 @@ pub enum ConvertCallInfoToExecuteInvocationError {
     GetFunctionInvocation(#[from] TryFuntionInvocationFromCallInfoError),
 }
 
+impl From<ConvertCallInfoToExecuteInvocationError> for StarknetRpcApiError {
+    fn from(err: ConvertCallInfoToExecuteInvocationError) -> Self {
+        match err {
+            ConvertCallInfoToExecuteInvocationError::TransactionExecutionFailed => StarknetRpcApiError::ContractError,
+            ConvertCallInfoToExecuteInvocationError::GetFunctionInvocation(_) => {
+                StarknetRpcApiError::InternalServerError
```
[the remark this answers is not in the document]
Comment (tdelabro) [reply crates/client/rpc/src/trace_api.rs:178]: We don't want to return to the RPC caller details about the internal implementation. We return the errors defined in starknet open-RPC, or `InternalServerError`. 
`InternalServerError` meaning is "we fucked up, sorry. It's us not you"

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,8 @@
 ## Next release
 
 - feat(rpc): added `get_state_update` real values from DA db
+- feat(rpc/trace_api): add `trace_block_transaction`
+- chore(runtime_api): remove unused `filter_extrinsic`
```
Comment (EvolveArt) [inline CHANGELOG.md:7]: add that you changed the hash in all DBs

File: starknet-rpc-test/Cargo.toml [hunk 3]
```diff
@@ -117,3 +117,7 @@ path = "simulate_transaction.rs"
 [[test]]
 name = "starknet_get_state_update"
 path = "get_state_update.rs"
+
+[[test]]
+name = "starknet_trance_block"
```
[the remark this answers is not in the document]
Comment (tdelabro) [reply starknet-rpc-test/Cargo.toml:122]: done

File: starknet-rpc-test/trace_block.rs [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        rpc.trace_block_transactions(BlockId::Hash(FieldElement::ZERO)).await,
+        Err(StarknetProviderError(StarknetErrorWithMessage { code: MaybeUnknownErrorCode::Known(code), .. })) if code == StarknetError::BlockNotFound
+    );
+
+    Ok(())
+}
+
+#[rstest]
+#[tokio::test]
+async fn work(madara: &ThreadSafeMadaraClient) -> Result<(), anyhow::Error> {
```
[the remark this answers is not in the document]
Comment (tdelabro) [reply starknet-rpc-test/trace_block.rs:34]: done

File: starknet-rpc-test/trace_block.rs [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    // Copy-pasted from add_invoke_transaction::work_with_storage_change
+    let funding_account = build_single_owner_account(&rpc, SIGNER_PRIVATE, ARGENT_CONTRACT_ADDRESS, true);
+    let recipient_account = FieldElement::from_hex_be("0x123").unwrap();
+
+    let block_number = {
+        let mut madara_write_lock = madara.write().await;
+
+        // Simulation at block 0 will fail for a reason I don't understand
+        // I think simulation may happen on the state before the genesis logic is executed
+        // meaning accounts don't have any funds yet
```
[the remark this answers is not in the document]
Comment (tdelabro) [reply starknet-rpc-test/trace_block.rs:47]: Sure

[the remark this answers is not in the document]
Comment (tdelabro) [reply starknet-rpc-test/trace_block.rs:47]: https://github.com/keep-starknet-strange/madara/issues/1443

**Review by antiyro — APPROVED** [review]

**Pull request merged**